### PR TITLE
Incoming / Outgoing Links

### PR DESCRIPTION
**Pull request by kenjinp**
# NEW! 🔗
list resources that have a relationship to this entity!

```typescript
// Links! 🔗🔗🔗🔗
const incomingLinks: PaginatedList<ResourceLinks> = await resourceInstance.getIncomingLinks({ from: 0, size: 20}: PaginationSettings);
const outgoingLinks: PaginatedList<ResourceLinks> = await resourceInstance.getOutgoingLinks({ from: 0, size: 20}: PaginationSettings);

/*
{
    ...paginationSettings,
    results: [{
        link: Resource | string, // Resource or URL
        isExternal: boolean, // does this resource exist on the platform, or a link to somewhere?
        predicate:  "https://myNexusInstance.com/vocabs/myOrg/friendProject/friend" // URL vocab link 
    }]
}
*/
```

### CODE REVIEW

File: src/Resource/utils.ts [hunk 1]
```diff
@@ -336,3 +340,50 @@ export async function deprecateSelfResource(
...
+
+export async function getIncomingLinks(
+  orgLabel: string,
+  projectLabel: string,
+  selfUrl: string,
+  paginationSettings: PaginationSettings,
+): Promise<PaginatedList<ResourceLink>> {
+  try {
+    const { from, size } = paginationSettings;
+    const view = await getSparqlView(orgLabel, projectLabel);
```
Comment (julienmachon) [inline src/Resource/utils.ts]: Can you use the static method `SarqlView.get` in order to avoid 1 api call?

File: README.md [hunk 2]
```diff
@@ -298,6 +298,9 @@ resourceInstance.update({
     context: { [field: string]: string };
     [field: string]: any;
   }): Promise<Resource>;
+
+// Links! 🔗🔗🔗🔗
+const incomingLinks: PaginatedList<ResourceLinks> = await resourceInstance.getIncomingLinks({ from: 0, size: 20}: PaginationSettings);
```
Comment (julienmachon) [inline README.md]: Are you going to expose a static method too? `Resource.getIncomingLinks`

File: src/Resource/utils.ts [hunk 3]
```diff
@@ -336,3 +340,50 @@ export async function deprecateSelfResource(
...
+      const total = +countResponse.results.bindings[0].total.value;
+      const queryResults = paginatedResponse.results.bindings;
+      const resourcePromises = queryResults.map(subjectPredicatePair => {
+        return Resource.getSelf(
+          subjectPredicatePair.s.value,
+          orgLabel,
+          projectLabel,
+        );
+      });
+      const resources = await Promise.all(resourcePromises);
```
Comment (julienmachon) [inline src/Resource/utils.ts]: :sweat_smile: this is hardcore. Do we have to fetch every single resources?

Comment (kenjinp) [reply src/Resource/utils.ts]: I'll do it anyways...

File: src/Resource/utils.ts (hunk 1 above)
Comment (kenjinp) [reply src/Resource/utils.ts]: They're the same

File: README.md (hunk 2 above)
Comment (kenjinp) [reply README.md]: yes!

**Comment by kenjinp** [comment]
⚠️ Important! 

I've changed the way `Resource.getSelf` works so it no longer requires `orgLabel` and `projectLabel` argument, but gets it directly from the provided `selfUrl` argument.

File: src/Resource/utils.ts [hunk 4]
```diff
@@ -336,3 +341,155 @@ export async function deprecateSelfResource(
...
+              FILTER(isIri(?o) && ?p NOT IN (nxv:updatedBy, nxv:createdBy, nxv:constrainedBy, nxv:project, rdf:type))
+            } .
+            OPTIONAL {
+              graph ?g2 {
+              	?o nxv:constrainedBy ?aSchema .
+                ?o nxv:self ?self
+              }
+            }
+      }
+} AS %resultSet
```
Comment (mdupont) [inline src/Resource/utils.ts]: alignment issue

Comment (kenjinp) [reply src/Resource/utils.ts]: I fought with this for a while and gave up 🤷‍♀️

**Pull request merged**